contrib/annotations/src/main/java/io/opentelemetry/contrib/annotations/WithSpan.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.contrib.annotations;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * This annotation marks that an execution of this method or constructor should result in a new
+ * {@link io.opentelemetry.trace.Span}.
+ *
+ * <p>Application developers can use this annotation to signal OpenTelemetry auto-instrumentation
+ * that a new span should be created whenever marked method is executed.
+ *
+ * @see <a href="https://github.com/open-telemetry/opentelemetry-auto-instr-java">OpenTelemetry
+ *     Auto-Instrumentation</a>
+ */
+@Target({ElementType.METHOD, ElementType.CONSTRUCTOR})
+@Retention(RetentionPolicy.RUNTIME)
+public @interface WithSpan {
+  /**
+   * Optional name of the created span.
+   *
+   * <p>If not specified, an appropriate default name will be created by auto-instrumentation.
```

Comment:
Wondering if we could add an example here, so users know what kind of names could be created.

Reply:
I decided not to do this on purpose. Not sure if it is a good idea to document here inner implementation details of auto-instr which may change in the future.

Reply:
Overall I'd like to see a super basic example of what names *might* be used. Otherwise, as a new user, I have **zero** idea of what name will appear.

Reply:
Fair enough. Will do.

Reply:
I think things like this is why I was hesitant to put this into the API in the first place. We could provide examples all day long in the javadoc on this annotation, but who knows what will actually be done by the auto-instrumentation agent, or some other annotation processor? I feel like putting examples in here could just confuse people even more when the actual span names end up looking different.

Reply:
@jkwatson Agreed but mentioning that X or Y name *might* be chosen would be a fine thing to mention (without giving away too many details). In that regard, a basic, single example of what it could look like would be more than enough.

So to not wait forever to have this PR merged, lets go ahead with what we have now @iNikem and *maybe* update the docs later on with a single, simple example.